### PR TITLE
RHTAPWATCH-244: Configure ClusterLogForwarder to Splunk for Production env

### PR DESCRIPTION
**Pull request by eisraeli**
This commit contains configuration for forwarding OCP application, infrastructure and audit logs to the internal RH Splunk. It also refactors some staging related configuration so it can be used more efficiently by Kustomize, so the staging environment might be affected as well by this change.

### CODE REVIEW

**Comment by eisraeli** [comment]
Hey @ifireball @Omeramsc @amisstea 
Can you please review this PR?
It will enable us to also retain production logs in Splunk.

File: components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml [hunk 1]
```diff
@@ -14,10 +14,33 @@ spec:
...
+    - name: splunk-receiver-application
+      secret:
+        name: log-forwarder-splunk-rhtap-application-secret
+      type: splunk
+      url: https://http-inputs-rhcorporate.splunkcloud.com
+    - name: splunk-receiver-audit
+      secret:
+        name: log-forwarder-splunk-rhtap-audit-secret
+      type: splunk
+      url: https://http-inputs-rhcorporate.splunkcloud.com
```
Comment (ifireball) [inline components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml:26]: putting configuration in base - while cleaner, will also apply it for DEV clusters - not suer we want that...

**Review by ifireball — APPROVED** [review]
/lgtm

But one small potential issue

File: components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml (hunk 1 above)
Comment (eisraeli) [reply components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml:26]: @ifireball We haven't added any configuration for dev so far so lets take care of the development environment at a later stage.  I'm not even sure if I can assume DEV clusters will have access to the Vault or HEC endpoints.

In the scope of this PR, the desired changes apply to staging and production.

Comment (ifireball) [reply components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml:26]: I'm not saying we should apply the configuration for DEV, I'm saying we need to make sure we're not applying it without intending to

File: components/monitoring/logging/staging/kustomization.yaml [hunk 2]
```diff
@@ -12,9 +12,21 @@ patches:
...
+    patch: |
+      - op: replace
+        path: /spec/dataFrom/0/extract/key
+        value: production/monitoring/logging/fluentd/splunk-forwarder-rhtap-staging-application
+  - target:
+      group: external-secrets.io
+      version: v1beta1
+      kind: ExternalSecret
+      name: log-forwarder-splunk-rhtap-audit-external-secret
+    patch: |
```
Comment (Omeramsc) [inline components/monitoring/logging/staging/kustomization.yaml:29]: can you explain how that works?

**Review by Omeramsc — APPROVED** [review]
LGTM

**Review by Omeramsc — APPROVED** [review]
LGTM

File: components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml (hunk 1 above)
Comment (eisraeli) [reply components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml:26]: I don't think we do as long as there is no `development` path.
There was also CloudWatch configuration before without any implicit development configuration.

File: components/monitoring/logging/staging/kustomization.yaml (hunk 2 above)
Comment (eisraeli) [reply components/monitoring/logging/staging/kustomization.yaml:29]: Are you asking about Kustomize's patching mechanism? 
Can you be more specific?

File: components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml (hunk 1 above)
Comment (eisraeli) [reply components/monitoring/logging/base/configure-logging/configure-logforwarder.yaml:26]: @ifireball I've created a `development` folder and added a `kustomization.yaml` file with no resources from base.
That way nothing will be created in the development environment.

File: components/monitoring/logging/staging/kustomization.yaml (hunk 2 above)
Comment (Omeramsc) [reply components/monitoring/logging/staging/kustomization.yaml:27]: Yes, I'm not familiar with the syntax

Comment (eisraeli) [reply components/monitoring/logging/staging/kustomization.yaml:27]: @Omeramsc It patches configuration which is in the base folder. 
If you wish, we can have a short meeting and I'll explain short the mechanism works.

**Review by amisstea — APPROVED** [review]

**Review by srivickynesh — APPROVED** [review]
/lgtm

**Pull request merged**